presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkFailure.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkFailure
+        extends RuntimeException
+{
+    private final String type;
+    private final String errorCode;
+
+    public PrestoSparkFailure(String type, String message, String errorCode, Throwable cause)
+    {
+        super(message, cause);
+        this.type = requireNonNull(type, "type is null");
+        this.errorCode = requireNonNull(errorCode, "errorCode is null");
+    }
+
+    public String getType()
+    {
+        return type;
+    }
+
+    public String getErrorCode()
+    {
+        return errorCode;
+    }
+
+    @Override
+    public String toString()
+    {
+        String message = getMessage();
+        if (message != null) {
+            return type + ": " + message;
+        }
+        return type;
+    }
+
+    public boolean isBroadcastJoinOOM()
+    {
+        return getErrorCode().equals("EXCEEDED_LOCAL_MEMORY_LIMIT")
+                && getMessage().contains("Query exceeded per-node broadcast memory limit");
+    }
```

Comment:
hmmmm this looks a bit hacky...

Can we return a set of `RetryExecutionStrategy` instead? Check my other comment at `toPrestoSparkFailure`